packages/rspack/src/loader-runner/index.ts

```diff
@@ -516,6 +517,12 @@ export async function runLoader(
 		return options;
 	};
 
+	let compilation: Compilation | undefined = compiler.compilation;
+	while (compilation) {
```

Comment:
NIT: I don't know if this impl is safe enough, it is better to add some step checks to avoid dead loop.